doc/rsp.md

```diff
@@ -280,95 +315,168 @@ mapped into the 16-bit lanes. Signed opcodes (`LPV`, `SPV`) map the value to bit
 value to bits `(14..7)`. Load instructions zero the bits outside the mapped range, while
 store instructions effectively ignore the other bits.
 
-The actual bytes accessed in DMEM start at `GPR[base] + (offset * 8)`, but they wrap around
-at the 64-bit (8-byte) boundary. `element` is used to select the first lane being accessed
-by the instruction (lane accesses also wrap around). 
+One way to think about the packed loads are as 128-bit loads that are then rotated based
+on alignment and element. After aligning the address `GPR[base] + (offset * 8)` to the
+64-bit boundary below, 128 bits starting from that boundary are read. Then, that 128-bit
+value is rotated left by the amount the address is unaligned, and right by `element`
+bytes. The leftmost 8 bytes are mapped to the appropriate bits of the 8 16-bit lanes.
+
+The packed stores generally behave as you would expect, mapping the appropriate bits of
+each lane to consecutive bytes in memory, starting with the lane specified by `element`.
+However, instead of wrapping at 8 lanes, packed stores wrap at 16, and change the mapping
+bits for "lanes" 8-15. `SPV` when a lane index is in the range `[8..15]` behaves like `SUV`
+when its lane index is in the range `[0..7]`, and vice versa.
 
 For instance:
 
-	// a0 is 64-bit aligned
-	LUV v1[e2],$06(a0)     // load bytes $06(a0)-$07(a0) into VPR[1]<2..3>, and then $00(a0)-$05(a0) into VPR[1]<4..7,0..1>
+    // a0 is 64-bit aligned
+    LUV v1[e5],$02(a0)     // load bytes $00(a0)-$04(a0) into VPR[1]<3..7>,
+                              and $0d(a0)-$0f(a0) into VPR[1]<0..2>
+    SUV v1[e5],$02(a0)     // write bytes $02(a0)-$04(a0) from VPR[1]<5..7> (14..7),
+                              and $05(a0)-$09(a0) from VPR[1]<0..5> (15..8)
 
-If `element` is higher than 8 (so, in the range `[8..15]`), load instructions wrap
-element access around (so basically the highest bit of `element` is ignored), while
-store instructions change the mapping bits, effectively swapping their meaning: `SPV`
-with `element` in the range `[8..15]` is like `SUV` with `element` in the range `[0..7]`,
-and viceversa.
```

Comment:
I think I prefer the previous version. Can you please explain me what you found difficult to follow? I think your version is closer to how you would implement it, but mine is close to describing how the hardware behaves.

Reply:
I think I read the original version as saying `SUV` would use the `(14..7)` mapping for all lanes, as long as the `element` value given in the instruction is in the range `[0..7]`. I wanted to make it clearer that when `element` is 1, the last lane will use the `(15..8)` mapping and the other lanes will use the `(14..7)` mapping.

I agree your load description is more clear. I tried using a "rotation" description because I thought some of the other instructions would be easier to describe after introducing the concept, but now that I see your `LTV` description I think it is possible to do without it.

Reply:
> I think I read the original version as saying SUV would use the (14..7) mapping for all lanes, as long as the element value given in the instruction is in the range [0..7].

You're right, this is not clear. Feel free to propose a new warning of that paragraph to clarify this.

Reply:
Actually, I think the part that I found difficult to describe without rotation is packed loads with `element`, like `LPV v0[e1] $00(a0)`, where lane 0 is loaded from `$0e(a0)`.

Reply:
I think this should be explained by my sentence "`element` is used to select the first lane being accessed by the instruction (lane accesses also wrap around)".

Maybe we should further clarify it with "`element` is used to select the first lane with `vt` which is loaded with the first 16-bit word at the specified memory address. Subsequent 16-bit values are loaded into subsequent lanes in the register, wrapping around within the register".

Reply:
Sorry, I misspoke. For packed loads specifically 16-bit lanes are loaded with consecutive 8-bit words, so `LPV v0[e1] $00(a0)` loads bytes `$0F,$00, $01,$02,$03,$04,$05,$06`. `$0F` is not subsequent to `$06`, and where this jump occurs depends on the load alignment. `LPV v0[e1] $01(a0)` is the same as `LPV v0[e0] $00(a0)`, for example.